module/userProjects/helper.js

```diff
@@ -2280,7 +2339,558 @@ module.exports = class UserProjectsHelper {
         })
     }
 
+    /**
+     * validate certificate criteria.
+     * @method
+     * @name criteriaValidation 
+     * @param {Object} data - project data for certificate creation 
+     * @returns 
+    */
+
+    static criteriaValidation(data) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                let criteria = data.certificate.criteria;
+                let validationResult = [];
+                let validationMessage = "";
+                let validationExpression = criteria.expression
+                if ( criteria.conditions &&  Object.keys(criteria.conditions).length > 0 ) {
+                    let conditions = criteria.conditions;
+                    let conditionKeys = Object.keys(conditions)
+
+                    for ( let index = 0; index < conditionKeys.length; index++ ) {
+                        // correntCondition contain the prefinal level data
+                        let currentCondition = conditions[conditionKeys[index]];
+
+                        //now pass expression and validation scope to another function which will start the validation procedure
+                        let validation = await _subCriteriaValidation( currentCondition.conditions, currentCondition.expression, data );
+                        
+                        validationResult.push(validation.success);
+                        ( validation.success == false ) ? validationMessage = validationMessage + " " + currentCondition.validationText : "";
+                    }
+                    let criteriaValidation = await _criteriaExpressionValidation( validationExpression, conditionKeys, validationResult )
+                    return resolve({
+                        success: criteriaValidation,
+                        message: ( criteriaValidation == false ) ? validationMessage : CONSTANTS.common.PROJECT_CERTIFICATE_GENERATED_SUCCESSFULLY
+                    });
+                }
+                return resolve({
+                    success: false
+                })
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: {}
+                });
+            }
+        })
+    }
+
+    /**
+     * generate project certificate.
+     * @method
+     * @name generateCertificate 
+     * @param {Object} data - project data for certificate creation data.
+     * @returns {JSON} certificate details.
+    */
+
+     static generateCertificate(data) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                //  if eligible key is not there check criteria for validation
+                if ( !data.certificate.eligible ) {
```

Comment:
@VISHNUDAS-tunerlabs we have to push to Kafka is certificate is required to be generated right? See the link - https://project-sunbird.atlassian.net/wiki/spaces/~5c249c50ab1c1d35825c3e26/pages/3176038408/Certificates+for+ML+Projects+capability+-+Implementation+Plan#Certificate-generation-for-user

Idea is write a consumer within ml project service listen to project topic, see if project is completed and is eligible for certificate, if yes push to another topic which will listen and call RC API.

Reply:
@aks30  checking certificate condition in consumer and calling certificate generation function if condition satisfied